drivers/SmartThings/zigbee-bed/src/shus-mattress/init.lua

```diff
@@ -0,0 +1,326 @@
+      cluster_base.write_manufacturer_specific_attribute(
+        device,
+        cluster.id,
+        attr.id,
+        cluster.mfg_specific_code,
+        attr.value_type,
+        attr.value.off
+      )
+    )
+  end
+end
+
+local function left_ai_mode_cap_handler(driver, device, cmd)
+  process_switch_cap(
+    device,
+    cmd.args.leftControl,
+    custom_clusters.shus_smart_mattress,
+    custom_clusters.shus_smart_mattress.attributes.left_ai_mode
+  )
+end
+
+local function right_ai_mode_cap_handler(driver, device, cmd)
+  process_switch_cap(
+    device,
+    cmd.args.rightControl,
+    custom_clusters.shus_smart_mattress,
+    custom_clusters.shus_smart_mattress.attributes.right_ai_mode
+  )
+end
+
+local function auto_inflation_cap_handler(driver, device, cmd)
+  process_switch_cap(
+    device,
+    cmd.args.stateControl,
+    custom_clusters.shus_smart_mattress,
+    custom_clusters.shus_smart_mattress.attributes.auto_inflation
+  )
+end
+
+local function strong_exp_mode_cap_handler(driver, device, cmd)
+  process_switch_cap(
+    device,
+    cmd.args.stateControl,
+    custom_clusters.shus_smart_mattress,
+    custom_clusters.shus_smart_mattress.attributes.strong_exp_mode
+  )
+end
+
+local function process_control_cap(device, value, cluster, attr)
+  device:send(
+    cluster_base.write_manufacturer_specific_attribute(
+      device,
+      cluster.id,
+      attr.id,
+      cluster.mfg_specific_code,
+      attr.value_type,
+      attr.value[value]
+    )
+  )
+end
+
+local function left_control_back_cap_handler(driver, device, cmd)
+  process_control_cap(
+    device,
+    cmd.args.backControl,
+    custom_clusters.shus_smart_mattress,
+    custom_clusters.shus_smart_mattress.attributes.left_back
+  )
+end
+
+local function left_control_waist_cap_handler(driver, device, cmd)
+  process_control_cap(
+    device,
+    cmd.args.waistControl,
+    custom_clusters.shus_smart_mattress,
+    custom_clusters.shus_smart_mattress.attributes.left_waist
+  )
+end
+
+local function left_control_hip_cap_handler(driver, device, cmd)
+  process_control_cap(
+    device,
+    cmd.args.hipControl,
+    custom_clusters.shus_smart_mattress,
+    custom_clusters.shus_smart_mattress.attributes.left_hip
+  )
+end
+
+local function right_control_back_cap_handler(driver, device, cmd)
+  process_control_cap(
+    device,
+    cmd.args.backControl,
+    custom_clusters.shus_smart_mattress,
+    custom_clusters.shus_smart_mattress.attributes.right_back
+  )
+end
+
+local function right_control_waist_cap_handler(driver, device, cmd)
+  process_control_cap(
+    device,
+    cmd.args.waistControl,
+    custom_clusters.shus_smart_mattress,
+    custom_clusters.shus_smart_mattress.attributes.right_waist
+  )
+end
+
+local function right_control_hip_cap_handler(driver, device, cmd)
+  process_control_cap(
+    device,
+    cmd.args.hipControl,
+    custom_clusters.shus_smart_mattress,
+    custom_clusters.shus_smart_mattress.attributes.right_hip
+  )
+end
```

Comment:
you could use a factory method here as well, just pass in "right_hip", "right_waist" etc in the handler declarations